EIPS/eip-9999.md

```diff
@@ -0,0 +1,285 @@
+- [Fork choice](https://github.com/ethereum/consensus-specs/blob/345e95504275d6c44b2bc251c50cb230fefb71b5/specs/_features/epbs/fork-choice.md) changes.
+- [P2P](https://github.com/ethereum/consensus-specs/blob/345e95504275d6c44b2bc251c50cb230fefb71b5/specs/_features/epbs/fork-choice.md) changes
+- [Honest validator guide](https://github.com/ethereum/consensus-specs/blob/345e95504275d6c44b2bc251c50cb230fefb71b5/specs/_features/epbs/fork-choice.md) changes.
+- A new [honest builder](https://github.com/ethereum/consensus-specs/blob/345e95504275d6c44b2bc251c50cb230fefb71b5/specs/_features/epbs/builder.md) guide. 
+- [Fork logic](TODO-add-link) changes. 
+
+A summary of the main changes is included below, the [Rationale](#rationale) section contains explanation for most of the design decisions around these changes. 
+
+#### Beacon chain changes
+
+##### Constants
+
+| Name | Value | 
+| - | - | 
+| `PAYLOAD_ABSENT` | `uint8(0)` |
+| `PAYLOAD_PRESENT` | `uint8(1)` | 
+| `PAYLOAD_WITHHELD` | `uint8(2)` | 
+| `PAYLOAD_INVALID_STATUS` | `uint8(3)` |
+
+##### Presets
+
+| Name | Value | 
+| - | - | 
+| `PTC_SIZE` | `uint64(2**9)` (=512) |
+| `DOMAIN_BEACON_BUILDER`     | `DomainType('0x1B000000')` |
+| `DOMAIN_PTC_ATTESTER`       | `DomainType('0x0C000000')` |
+| `MAX_PAYLOAD_ATTESTATIONS` | `2**2` (= 4) |
+ 
+##### Containers
+
+```python
+class PayloadAttestationData(Container):
+    beacon_block_root: Root
+    slot: Slot
+    payload_status: uint8
+```
+
+```python
+class PayloadAttestation(Container):
+    aggregation_bits: Bitvector[PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+```python
+class PayloadAttestationMessage(Container):
+    validator_index: ValidatorIndex
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+```python
+class IndexedPayloadAttestation(Container):
+    attesting_indices: List[ValidatorIndex, PTC_SIZE]
+    data: PayloadAttestationData
+    signature: BLSSignature
+```
+
+```python
+class SignedExecutionPayloadHeader(Container):
+    message: ExecutionPayloadHeader
+    signature: BLSSignature
+```
+    
+```python
+class ExecutionPayloadEnvelope(Container):
+    payload: ExecutionPayload
+    builder_index: ValidatorIndex
+    beacon_block_root: Root
+    blob_kzg_commitments: List[KZGCommitment, MAX_BLOB_COMMITMENTS_PER_BLOCK]
+    payload_withheld: boolean
+    state_root: Root
+```
+
+```python
+class SignedExecutionPayloadEnvelope(Container):
+    message: ExecutionPayloadEnvelope
+    signature: BLSSignature
+```
+
+The `BeaconState` container is modified with the addition of:
+- `latest_block_hash`, of type `Hash32`, to track the blockhash of the last execution payload in the blockchain. 
+- `latest_full_slot`, of type `Slot`, to track the latest slot in which an execution payload was included. 
+- `latest_widthdrawals_root` of type `Root` to track the hash tree root of the latests withdrawals deducted in the beacon chain when processing a `SignedBeaconBlock`. 
+
```

Comment:
Should we mention `BeaconBlockBody` is modified with the addition of aggregated payload attestations?